[docs-logging]: move developer/architecture docs to user docs (#119125)
elastic/kibana: docs/settings/logging-settings.asciidoc

{kib} relies on three high-level entities to set the logging service: appenders, loggers, and root. These can be configured in the `logging` namespace in `kibana.yml`.

- Appenders define where log messages are displayed (stdout or console) and their layout (`pattern` or `json`). They also allow you to specify if you want the logs stored and, if so, where (file on the disk).
- Loggers define what logging settings, such as the level of verbosity and the appenders, to apply to a particular context. Each log entry context provides information about the service or plugin that emits it and any of its sub-parts, for example, `metrics.ops` or `elasticsearch.query`.
- Root is a logger that applies to all the log entries in {kib}.

The valid settings in the `logging.appenders` namespace vary depending on the appender type.
To learn more about {kib} logging configuration, go to <<logging-configuration, Configure Logging in {kib}>>.

Refer to the <<log-settings-examples, examples>> for common configuration use cases. 

[cols="2*<"]
|===

> | `logging.appender.<appender-name>`

Author: I followed the format `security` docs use as a [placeholder](https://www.elastic.co/guide/en/kibana/master/security-settings-kb.html#authentication-provider-settings).

++++

{kib} relies on three high-level entities to set the logging service: appenders, loggers, and root.
You do not need to configure any additional settings to use the logging features in {kib}. Logging is enabled by default and will log at `info` level with a `pattern` layout.

NOTE: The logging configuration is validated against the predefined schema and if there are any issues with it, {kib} will fail to start with the detailed error message.

{kib} relies on three high-level entities to set the logging service: appenders, loggers, and root. These can be configured in the `logging` namespace in `kibana.yml`.

- Appenders define where log messages are displayed (stdout or console) and their layout (`pattern` or `json`). They also allow you to specify if you want the logs stored and, if so, where (file on the disk).
- Loggers define what logging settings, such as the level of verbosity and the appenders, to apply to a particular context. Each log entry context provides information about the service or plugin that emits it and any of its sub-parts, for example, `metrics.ops` or `elasticsearch.query`.
- Root is a logger that applies to all the log entries in {kib}.

> Refer to the <<log-settings-examples, examples>> for common configuration use cases. To learn more about possible configuration values, go to {kibana-ref}/logging-service.html[{kib}'s Logging service].

Author: I moved all of this with some changes to a logging configuration guide.

> The folowing table serves as a quick reference for different logging configuration keys. Note that these are not stand-alone settings and may require additional logging configuration. See the <<logging-configuration, Configure Logging in {kib}>> guide and complete <<log-settings-examples, examples>> for common configuration use cases. 

Author: This page is a quick reference look-up table without too many details. Logging configuration depends on the appender configuration and for complete guidance, has moved to the logging-configuration guide.

[cols="2*<"]
|===
| `logging.appenders[].<appender-name>`
| Unique appender identifier.

> | `logging.appenders[].console:`

Author: I used a similar format for indicating an array as maps used in [7.16](https://www.elastic.co/guide/en/kibana/7.16/settings.html), i.e. `map.regionmap.layers[].attribution`.

| Appender to use for logging records to *stdout*. By default, uses the `[%date][%level][%logger] %message` **pattern** layout. To use a **json**, set the <<log-in-json-ECS-example,layout type to `json`>>.

| `logging.appenders[].file:`

> | Allows you to specify a fileName to send log records to on disk. To send <<log-to-file-example,all log records to file>>, add the file appender to `root.appenders`. If configured, you also need to specify <<log-to-file-example, `logging.appenders.file.pathName`>>.

Reviewer: ```suggestion
| Allows you to specify a fileName to write log records to disk. To write <<log-to-file-example,all log records to file>>, add the file appender to `root.appenders`. If configured, you also need to specify <<log-to-file-example, `logging.appenders.file.pathName`>>.
```

| `logging.appenders[].rolling-file:`
| Similar to Log4j's `RollingFileAppender`, this appender will log into a file and rotate if following a rolling strategy when the configured policy triggers. There are currently two policies supported: `size-limit` and `time-interval`.

| `logging.appenders[].<appender-name>.type`

> | Determines where the log messages are output. Options are `console`, `file`, `rewrite`, `rolling-file`. Required.

Reviewer: ```suggestion
| The appender type determines where the log messages are sent. Options are `console`, `file`, `rewrite`, `rolling-file`. Required.
```

below we talk about "file and rolling-file appender types" so thought using the same language might make it a bit clearer that this is where you choose the type.

| `logging.appenders[].<appender-name>.fileName`
| Determines the filepath where the log messages are written to for file and rolling-file appender types. Required for appenders that write to file.

| `logging.appenders[].<appender-name>.policy.type`
| Specify the triggering policy for when a rollover should occur for the `rolling-file` type appender. 

| `logging.appenders[].<appender-name>.policy.interval`
| Specify the time interval for rotating a log file for a `time-interval` type `rolling-file` appender. *Default 24h*

| `logging.appenders[].<appender-name>.policy.size`
| Specify the size limit at which the policy should trigger a rollover for a `size-limit` type `rolling-file` appender. *Default 100mb*. 

| `logging.appenders[].<appender-name>.policy.interval`
| Specify the time interval at which the policy should trigger a rollover for a time-interval type `rolling-file` appender. 

| `logging.appenders[].<appender-name>.policy.modulate`

> | Whether the interval should be adjusted to cause the next rollover to occur on the interval boundary. Boolean.

Reviewer: ```suggestion
| Whether the interval should be adjusted to cause the next rollover to occur on the interval boundary. Boolean. Default `true`.
```

| `logging.appenders[].<appender-name>.strategy.type`
| Rolling file strategy type. Only `numeric` is currently supported.

| `logging.appenders[].<appender-name>.strategy.pattern`
| The suffix to append to the file path when rolling. Must include `%i`.

| `logging.appenders[].<appender-name>.strategy.max`
| The maximum number of files to keep. Optional. Default is `7`.

| `logging.appenders[].<appender-name>.layout.type`

> | Determines how the log messages are displayed. Options are `pattern` or `json`. Required.

Reviewer 2: ```suggestion
| Determines how the log messages are displayed. Options are `pattern`, which provides human-readable output, or `json`, which provides ECS-compliant output. Required.
```

| `logging.appenders[].<appender-name>.layout.highlight`

> | Optional boolean for to highlight log messages in color. Applies to `pattern` layout only.

Reviewer: ```suggestion
| Optional boolean to highlight log messages in color. Applies to `pattern` layout only. Default is `false`.
```

Reviewer: Almost corrected your americanised "color" 🇬🇧  😂

| `logging.appenders[].<appender-name>.layout.pattern`
| Optional <<pattern-layout, string pattern>> for placeholders that will be replaced with data from the actual log message. Applicable to pattern type layout only.

| `logging.root.appenders[]`
| List of specific appenders to apply to `root`. Defaults to `console` with `pattern` layout.

| `logging.root.level`

> | Specify default verbosity for all log messages to fall back to if not specifically configured at the individual logger level. Options are all, fatal, error, warn, info, debug, trace, off. The all and off levels can be used only in configuration and are just handy shortcuts that allow you to log every log record or disable logging entirely or for a specific logger. Default is `info`.

Reviewer: ```suggestion
| Specify default verbosity for all log messages to fall back to if not specifically configured at the individual logger level. Options are `all`, `fatal`, `error`, `warn`, `info`, `debug`, `trace`, `off`. The `all` and `off` levels can be used only in configuration and are just handy shortcuts that allow you to log every log record or disable logging entirely or for a specific logger. Default is `info`.
```

Use backticks for actual values for consistency with other examples.

| `logging.loggers[].<logger>.name:`
| Specific logger instance.

| `logging.loggers[].<logger>.level`

> | Specify verbosity of log messages for <logger> context. Optional and falls back to `root` level if not configured.

Reviewer: ```suggestion
| Specify verbosity of log messages for <logger> context. Optional and falls back to the verbosity of the `root` logger if not configured.
```

I think this should be _root logger_ not _root level_ correct?

Reviewer: Reading the other docs, seems like this should be "inherits the verbosity of any ancestor logger, up to the `root` logger"?

Author: Good point! I've changed the wording to your recommendation.

> | Specify verbosity of log messages for <logger> context. Optional and falls back to `root` level if not configured.

Author: ```suggestion
| Specify verbosity of log messages for <logger> context. Optional and inherits the verbosity of any ancestor logger, up to the `root` logger `level`.
```

| `logging.loggers[].<logger>.appenders:`
| Specific appenders to apply for a particular logger context.

| `logging.loggers[].<logger>.appenders`

> | Determines the appender to apply to a specific logger context as an array. Optional and falls back to the `root` appender(s) if not specified.

Reviewer: ```suggestion
| Determines the appender to apply to a specific logger context as an array. Optional and falls back to the appender(s) of the `root` logger if not specified.
```

> | `logging.loggers[].<logger>.appenders:`
> | Specific appenders to apply for a particular logger context.
> 
> | `logging.loggers[].<logger>.appenders`
> | Determines the appender to apply to a specific logger context as an array. Optional and falls back to the appender(s) of the `root` logger if not specified.

Reviewer 2: duplicate entry

```suggestion
| `logging.loggers[].<logger>.appenders`
| Determines the appender to apply to a specific logger context as an array. Optional and falls back to the appender(s) of the `root` logger if not specified.
```
